bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/Caret.java

```diff
@@ -414,6 +414,14 @@ void setBoundsInPixels (int x, int y, int width, int height) {
 		resized = true;
 		if (isVisible && hasFocus ()) resize ();
 	}
+	setBoundsInPixels(DPIUtil.autoScaleUp(x, zoom), DPIUtil.autoScaleUp(y, zoom), sameExtent);
+}
+
+private void setBoundsInPixels (int xInPixels, int yInPixels, boolean sameExtent) {
```

Comment:
The method name is misleading. It does not set the bounds but only the x/y values (not width/height).